docs/rules/prefer-switch.md

```diff
@@ -0,0 +1,120 @@
+# Prefer `switch` over multiple `else-if`
+
+A switch statement is easier to read than multiple if statements with simple equality comparisons.
+
+This rule is partly fixable.
+
+## Fail
+
+```js
+if (foo === 1) {
+	// 1
+} else if (foo === 2) {
+	// 2
+} else if (foo === 3) {
+	// 3
+} else {
+	// default
+}
+```
+
+## Pass
+
+```js
+if (foo === 1) {
+	// 1
+} else if (foo === 2) {
+	// 2
+}
+```
+
+```js
+switch (foo) {
+	case 1: {
+		// 1
+		break;
+	}
+	case 2: {
+		// 2
+		break;
+	}
+	case 3: {
+		// 3
+		break;
+	}
+	default: {
+		// default
+	}
+}
+```
+
+### `options`
+
+Type: `object`
+
+#### `minimumCases`
+
+Type: `integer`\
+Minimum: `2`\
+Default: `3`
+
+The minimum number of cases before reporting.
+
+The `default` branch doesn't count. Multiple comparisons on the same `if` block is considered one case.
+
+Examples:
+
+```js
+// eslint unicorn/prefer-switch: ["error", {"minimumCases": 4}]
+if (foo === 1) {}
+else (foo === 2) {}
+else (foo === 3) {}
+
+// Passes, only 3 cases.
+```
+
+```js
+// eslint unicorn/prefer-switch: ["error", {"minimumCases": 4}]
+if (foo === 1) {}
+else (foo === 2) {}
+else (foo === 3) {}
+else {}
+
+// Passes, only 3 cases.
+```
+
+```js
+// eslint unicorn/prefer-switch: ["error", {"minimumCases": 4}]
+if (foo === 1) {}
+else if (foo === 2 || foo === 3) {}
+else if (foo === 4) {}
+
+// Passes, only 3 cases.
+```
+
+```js
+// eslint unicorn/prefer-switch: ["error", {"minimumCases": 2}]
+if (foo === 1) {}
+else if (foo === 2) {}
+
+// Fails
+```
+
+#### `emptyDefaultCase`
+
+Type: `string`\
+Default: `'no-default-comment'`
+
+- `'no-default-comment'` (default)
+	Insert `// No default` comment after last case.
```

Comment:
I don't think it's clear to the user what the comments do. It should refer to https://eslint.org/docs/rules/default-case